src/promptflow/promptflow/_sdk/entities/_run.py

```diff
@@ -196,18 +202,32 @@ def properties(self) -> Dict[str, str]:
                 result[FlowRunProperties.RUN] = run_name
             if self.variant:
                 result[FlowRunProperties.NODE_VARIANT] = self.variant
-            return {
-                **result,
-                **self._properties,
+        elif self._run_source == RunInfoSources.EXISTING_RUN:
+            result = {
+                FlowRunProperties.OUTPUT_PATH: Path(self.source).resolve().as_posix(),
             }
-        return self._properties
+
+        return {
+            **result,
+            **self._properties,
+        }
 
     @classmethod
     def _from_orm_object(cls, obj: ORMRun) -> "Run":
         properties_json = json.loads(str(obj.properties))
+        flow = properties_json.get(FlowRunProperties.FLOW_PATH, None)
+
+        # there can be two sources for orm run object:
+        # 1. LOCAL: Created when run is created from local flow
+        # 2. EXISTING_RUN: Created when run is created from existing run folder
+        source = None
+        if getattr(obj, "run_source", None) == RunInfoSources.EXISTING_RUN:
+            source = properties_json[FlowRunProperties.OUTPUT_PATH]
+
         return Run(
             name=str(obj.name),
-            flow=Path(properties_json[FlowRunProperties.FLOW_PATH]),
+            flow=Path(flow) if flow else None,
```

Comment:
Can we directly use snapshot path as flow path when flow's from source?

Reply:
In content wise they are equal, but in concept wise the flow path is the flow that the run was created against and snapshot is a copy of that flow. Pointing the flow path to a snapshot folder is kinda twisted, so I'm not sure if we should do that.